--- /ConfigureChecks.cmake	2023-11-28 16:13:15
+++ /ConfigureChecks.cmake	2023-11-28 15:58:45
@@ -37,13 +37,7 @@
 #/* Define to 1 if you have the ANSI C header files. */
 check_include_files("stdlib.h;stdarg.h;string.h;float.h" STDC_HEADERS)
 
-test_big_endian(WORDS_BIGENDIAN)
-#/* 1234 = LIL_ENDIAN, 4321 = BIGENDIAN */
-if(WORDS_BIGENDIAN)
-    set(BYTEORDER 4321)
-else(WORDS_BIGENDIAN)
-    set(BYTEORDER 1234)
-endif(WORDS_BIGENDIAN)
+set(BYTEORDER 1234)
 
 if(HAVE_SYS_TYPES_H)
     check_symbol_exists("off_t" "sys/types.h" OFF_T)
