--- /CMakeLists.txt	2024-05-23 13:22:05
+++ /CMakeLists.txt	2024-05-23 13:26:44
@@ -452,8 +452,8 @@
 endif ()
 
 # The list of tools (to be installed into, e.g., /usr/local/bin)
-set (TOOLS CartConvert ConicProj GeodesicProj GeoConvert GeodSolve
-  GeoidEval Gravity MagneticField Planimeter RhumbSolve TransverseMercatorProj)
+#set (TOOLS CartConvert ConicProj GeodesicProj GeoConvert GeodSolve
+#  GeoidEval Gravity MagneticField Planimeter RhumbSolve TransverseMercatorProj)
 # The list of scripts (to be installed into, e.g., /usr/local/sbin)
 set (SCRIPTS geographiclib-get-geoids geographiclib-get-gravity
   geographiclib-get-magnetic)
@@ -463,7 +463,7 @@
 # The list of subdirectories to process
 add_subdirectory (src)
 add_subdirectory (include/GeographicLib)
-add_subdirectory (tools)
+#add_subdirectory (tools)
 add_subdirectory (man)
 add_subdirectory (doc)
 if (EXAMPLEDIR)