diff --git a/port/cpl_vsi.h b/port/cpl_vsi.h
index 80c66795a3..47751df4f1 100644
--- a/port/cpl_vsi.h
+++ b/port/cpl_vsi.h
@@ -719,6 +719,13 @@ void CPL_DLL VSIFreeFilesystemPluginCallbacksStruct(
 int CPL_DLL VSIInstallPluginHandler(
     const char *pszPrefix, const VSIFilesystemPluginCallbacksStruct *poCb);
 
+/**
+ * Unregister a handler previously installed with VSIInstallPluginHandler() on
+ * the given prefix.
+ * @since GDAL 3.9
+ */
+int CPL_DLL VSIRemovePluginHandler(const char *pszPrefix);
+
 /* ==================================================================== */
 /*      Time querying.                                                  */
 /* ==================================================================== */
diff --git a/port/cpl_vsi_virtual.h b/port/cpl_vsi_virtual.h
index 7c9b9ca304..bf41ee6bbd 100644
--- a/port/cpl_vsi_virtual.h
+++ b/port/cpl_vsi_virtual.h
@@ -327,8 +327,7 @@ class CPL_DLL VSIFileManager
     static VSIFilesystemHandler *GetHandler(const char *);
     static void InstallHandler(const std::string &osPrefix,
                                VSIFilesystemHandler *);
-    /* RemoveHandler is never defined. */
-    /* static void RemoveHandler( const std::string& osPrefix ); */
+    static void RemoveHandler(const std::string& osPrefix);
 
     static char **GetPrefixes();
 };
diff --git a/port/cpl_vsil.cpp b/port/cpl_vsil.cpp
index 3a649d50bf..74683b840d 100644
--- a/port/cpl_vsil.cpp
+++ b/port/cpl_vsil.cpp
@@ -3253,6 +3253,18 @@ void VSIFileManager::InstallHandler(const std::string &osPrefix,
         Get()->oHandlers[osPrefix] = poHandler;
 }
 
+/************************************************************************/
+/*                          RemoveHandler()                             */
+/************************************************************************/
+
+void VSIFileManager::RemoveHandler(const std::string& osPrefix)
+{
+    if (osPrefix == "")
+        Get()->poDefaultHandler = nullptr;
+    else
+        Get()->oHandlers.erase(osPrefix);
+}
+
 /************************************************************************/
 /*                       VSICleanupFileManager()                        */
 /************************************************************************/
diff --git a/port/cpl_vsil_plugin.cpp b/port/cpl_vsil_plugin.cpp
index 945f878f55..f35d0aaf74 100644
--- a/port/cpl_vsil_plugin.cpp
+++ b/port/cpl_vsil_plugin.cpp
@@ -470,6 +470,12 @@ int VSIInstallPluginHandler(const char *pszPrefix,
     return 0;
 }
 
+int VSIRemovePluginHandler(const char *pszPrefix)
+{
+    VSIFileManager::RemoveHandler(pszPrefix);
+    return 0;
+}
+
 VSIFilesystemPluginCallbacksStruct *
 VSIAllocFilesystemPluginCallbacksStruct(void)
 {
